Modules/Test/classes/class.ilObjTestSettingsScoringResultsGUI.php

```diff
@@ -127,43 +179,60 @@ private function confirmedSaveFormCmd()
 
     private function saveFormCmd($isConfirmedSave = false)
     {
-        $form = $this->buildForm();
-
-        // form validation and initialisation
-
-        $errors = !$form->checkInput(); // ALWAYS CALL BEFORE setValuesByPost()
-        $form->setValuesByPost(); // NEVER CALL THIS BEFORE checkInput()
+        $form = $this->buildForm()
+            ->withRequest($this->request);
 
-        // return to form when any form validation errors exist
-
-        if ($errors) {
+        $result = $form->getInputGroup()->getContent();
```

Comment:
Please just use `getData` on the `$form` here. It will either return `null` when input is faulty or the actual result if that is not the case. `getInputGroup` is part of the internals only.